invenio_rdm_records/resources/serializers/crossref/__init__.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021-2024 CERN.
+# Copyright (C) 2025 Front Matter.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Crossref Serializers for Invenio RDM Records."""
+
+from commonmeta import (
+    MARSHMALLOW_MAP,
+    CrossrefError,
+    CrossrefXMLSchema,
+    Metadata,
+    convert_crossref_xml,
+    unparse_xml,
+)
+from flask import current_app
+from flask_resources import BaseListSchema, MarshmallowSerializer
+from flask_resources.serializers import SimpleSerializer
+
+
+class CrossrefXMLSerializer(MarshmallowSerializer):
+    """JSON based Crossref XML serializer for records."""
+
+    def __init__(self, **options):
+        """Constructor."""
+        super().__init__(
+            format_serializer_cls=SimpleSerializer,
+            object_schema_cls=CrossrefXMLSchema,
+            list_schema_cls=BaseListSchema,
+            encoder=self.crossref_xml_tostring,
+            **options,
+        )
+
+    def dump_obj(self, record):
```

Comment:
This whole function looks not really as a invenio serializer looks like. for me it is to much going on in the `dump_obj` method.

would it be possible to move this stuff into the `CrossrefXMLSchema` or a subclass created in rdm-records like `class RDMCrossrefXMLSchema(CrossrefXMLSChema):`

in my thinking it should not be necessary to override the `dump_obj` method. it `schema.dump` should do the magick and return a xml if that is necessary.

Reply:
I simplified the dump_obj method, using the write_crossref_xml method from commonmeta-py instead. What makes serialization complicated is that Crossref XML has a different field order depending on content type, therefore the  field order has to be changed depending on type before XML serialization.